back/src/Guide/Markdown.hs

```diff
@@ -320,6 +341,19 @@ instance Aeson.ToJSON MarkdownTree where
   toJSON md = Aeson.object [
     "text" Aeson..= markdownTreeSource md ]
 
+instance Aeson.FromJSON MarkdownInline where
+  parseJSON = Aeson.withObject "MarkdownInline" $ \o -> do
+    txt <- o Aeson..: "text"
+    pure $ toMarkdownInline txt
+instance Aeson.FromJSON MarkdownBlock where
+  parseJSON = Aeson.withObject "MarkdownBlock" $ \o -> do
+    txt <- o Aeson..: "text"
+    pure $ toMarkdownBlock txt
+instance Aeson.FromJSON MarkdownTree where
+  parseJSON = Aeson.withObject "MarkdownTree" $ \o -> do
+    txt <- o Aeson..: "text"
+    pure $ toMarkdownTree "" txt
```

Comment:
You should also get `markdownTreeIdPrefix`. This is an omission in the `ToJSON` instance I wrote, it should be there as well.